tests/interfaces/test_jax.py

```diff
@@ -862,6 +834,230 @@ def cost(a, cache):
+                qml.RX(a[1], wires=0)
+                qml.RY(a[2], wires=0)
+
+                for r in ret:
+                    qml.apply(r)
+
+            res = qml.interfaces.execute(
+                [tape], dev, cache=cache, interface="jax-jit", **execute_kwargs
+            )[0]
+            return res
+
+        res = cost(params, cache=None)
+        assert res.shape == out_dim
+
+    def test_qnode_sample(self, execute_kwargs):
+        """Tests computing multiple expectation values in a tape."""
+        dev = qml.device("default.qubit", wires=2, shots=10)
+        params = jnp.array([0.1, 0.2, 0.3])
+
+        grad_meth = (
+            execute_kwargs["gradient_kwargs"]["method"]
+            if "gradient_kwargs" in execute_kwargs
+            else ""
+        )
+        if "adjoint" in grad_meth or "backprop" in grad_meth:
+            pytest.skip("Adjoint does not support probs")
+
+        def cost(a, cache):
+            with qml.tape.QuantumTape() as tape:
+                qml.RY(a[0], wires=0)
+                qml.RX(a[1], wires=0)
+                qml.RY(a[2], wires=0)
+                qml.sample(qml.PauliZ(0))
+
+            res = qml.interfaces.execute(
+                [tape], dev, cache=cache, interface="jax-jit", **execute_kwargs
+            )[0]
+            return res
+
+        res = cost(params, cache=None)
+        assert res.shape == (1, dev.shots)
+
+    def test_multiple_expvals_grad(self, execute_kwargs):
+        """Tests computing multiple expectation values in a tape."""
+        dev = qml.device("default.qubit", wires=2)
+        params = jnp.array([0.1, 0.2, 0.3])
+        fwd_mode = execute_kwargs.get("mode", "not forward") == "forward"
+        if fwd_mode:
+            pytest.skip("The forward mode is tested separately as it should raise an error.")
+
+        def cost(a, cache):
+            with qml.tape.QuantumTape() as tape:
+                qml.RY(a[0], wires=0)
+                qml.RX(a[1], wires=0)
+                qml.RY(a[2], wires=0)
+                qml.expval(qml.PauliZ(0))
+                qml.expval(qml.PauliZ(1))
+
+            res = qml.interfaces.execute(
+                [tape], dev, cache=cache, interface="jax-jit", **execute_kwargs
+            )[0]
+            return res[0] + res[1]
+
+        res = jax.grad(cost)(params, cache=None)
+        assert res.shape == (3,)
+
+    def test_multi_tape_jacobian_probs_expvals(self, execute_kwargs):
+        """Test the jacobian computation with multiple tapes with probability
+        and expectation value computations."""
+        fwd_mode = execute_kwargs.get("mode", "not forward") == "forward"
+        if fwd_mode:
+            pytest.skip("The forward mode is tested separately as it should raise an error.")
+
+        adjoint = execute_kwargs.get("gradient_kwargs", {}).get("method", "") == "adjoint_jacobian"
+        if adjoint:
+            pytest.skip("The adjoint diff method doesn't support probabilities.")

```

Comment:
Sure :) We're parametrizing over the differentiation methods in this test case and this logic skips the test case when we have adjoint differentiation. We're skipping the testing because adjoint doesn't support `qml.probs` (so we'd have an error raised otherwise).

`adjoint = execute_kwargs.get("gradient_kwargs", {}).get("method", "") == "adjoint_jacobian"` is the logic that looks scary and determines if we have adjoint defined as the diff method.

Reply:
Thanks! This totally make sense.